tests/suites/0_stateless/03_dml/03_0020_group_by_negative.sql

```diff
@@ -1,27 +1,27 @@
+SELECT (0 - number) :: Int16 AS c FROM numbers(65535) group by (0 - number) :: Int16  HAVING (0 - number) :: Int16 = -1;
+SELECT (0 - number) :: Int16 AS c FROM numbers(65535) group by (0 - number) :: Int16  HAVING (0 - number) :: Int16 = -32768;
+SELECT (0 - number) :: Int16 AS c FROM numbers(65535) group by (0 - number) :: Int16  HAVING (0 - number) :: Int16 = -32769;
+SELECT COUNT() FROM (SELECT (0 - number) :: Int16 AS c FROM numbers(65535) group by (0 - number) :: Int16 );
+SELECT COUNT() FROM (SELECT (0 - number) :: Int16 AS c FROM numbers(65535) group by (0 - number) :: Int16  HAVING (0 - number) :: Int16 > 0);
 
-SELECT (0 - number) :: Int32 AS c FROM numbers(65535) GROUP BY c HAVING c = 0;
-SELECT (0 - number) :: Int32 AS c FROM numbers(65535) GROUP BY c HAVING c = -1;
-SELECT (0 - number) :: Int32 AS c FROM numbers(65535) GROUP BY c HAVING c = -32768;
-SELECT (0 - number) :: Int32 AS c FROM numbers(65535) GROUP BY c HAVING c = -32769;
-SELECT COUNT() FROM (SELECT (0 - number) :: Int32 AS c FROM numbers(65535) GROUP BY c);
-SELECT COUNT() FROM (SELECT (0 - number) :: Int32 AS c FROM numbers(65535) GROUP BY c HAVING c > 0);
+SELECT (0 - number) :: Int32 AS c FROM numbers(65535) group by (0 - number) :: Int32  HAVING (0 - number) :: Int32 = 0;
+SELECT (0 - number) :: Int32 AS c FROM numbers(65535) group by (0 - number) :: Int32  HAVING (0 - number) :: Int32 = -1;
+SELECT (0 - number) :: Int32 AS c FROM numbers(65535) group by (0 - number) :: Int32  HAVING (0 - number) :: Int32 = -32768;
+SELECT (0 - number) :: Int32 AS c FROM numbers(65535) group by (0 - number) :: Int32  HAVING (0 - number) :: Int32 = -32769;
+SELECT COUNT() FROM (SELECT (0 - number) :: Int32 AS c FROM numbers(65535) group by (0 - number) :: Int32 );
+SELECT COUNT() FROM (SELECT (0 - number) :: Int32 AS c FROM numbers(65535) group by (0 - number) :: Int32  HAVING (0 - number) :: Int32 > 0);
 
-SELECT (0 - number) :: Int64 AS c FROM numbers(65535) GROUP BY c HAVING c = 0;
-SELECT (0 - number) :: Int64 AS c FROM numbers(65535) GROUP BY c HAVING c = -1;
-SELECT (0 - number) :: Int64 AS c FROM numbers(65535) GROUP BY c HAVING c = -32768;
-SELECT (0 - number) :: Int64 AS c FROM numbers(65535) GROUP BY c HAVING c = -32769;
-SELECT COUNT() FROM (SELECT (0 - number) :: Int64 AS c FROM numbers(65535) GROUP BY c);
-SELECT COUNT() FROM (SELECT (0 - number) :: Int64 AS c FROM numbers(65535) GROUP BY c HAVING c > 0);
+SELECT (0 - number) :: Int64 AS c FROM numbers(65535) group by (0 - number) :: Int64  HAVING (0 - number) :: Int64 = 0;
+SELECT (0 - number) :: Int64 AS c FROM numbers(65535) group by (0 - number) :: Int64  HAVING (0 - number) :: Int64 = -1;
+SELECT (0 - number) :: Int64 AS c FROM numbers(65535) group by (0 - number) :: Int64  HAVING (0 - number) :: Int64 = -32768;
+SELECT (0 - number) :: Int64 AS c FROM numbers(65535) group by (0 - number) :: Int64  HAVING (0 - number) :: Int64 = -32769;
```

Comment:
Is it mean that the alias can't apply to the group by expression in the new planner?

Reply:
@sundy-li Yes, it follows the behavior of postgres. It will implicitly change the semantic of the `SELECT` clause, which may potentially make the code unclear. Thus we decided to support it later.

Reply:
Ok, seems postgres supports this:
https://www.db-fiddle.com/f/d9MhfGFV8vU9Ywom1zN4ve/0

Reply:
I remember we discussed it some times ago.
But group alias is very useful for the user,I think most databases support it.

Reply:
> Ok, seems postgres supports this: https://www.db-fiddle.com/f/d9MhfGFV8vU9Ywom1zN4ve/0

Yes, we support it too. But you cannot use `c` in `HAVING` clause.

> I remember we discussed it some time ago. But group alias is very useful for the user, I think most databases (should) support it.

We have supported group by aliased item, it won't take much effort to support referencing the alias in `HAVING` clause. I've just opened a issue for this https://github.com/datafuselabs/databend/issues/5660.

Reply:
Here's the PR https://github.com/datafuselabs/databend/pull/5662